packaging/linux/arch/arch_common.sh

```diff
@@ -0,0 +1,63 @@
+#! /usr/bin/env bash
+
+clone_maybe() {
+  if [ ! -e "$2" ] ; then
+    git clone "$1" "$2"
+  fi
+}
+
+setup_arch_build() {
+    build_root="$1"
+    git_url="$2"
+    src_prefix="$3"
+
+    # build_binaries.sh and deb/package_binaries.sh must already have
+    # been run on this build root.
+    here="$(dirname "$BASH_SOURCE")"
+
+    mkdir -p "$build_root/arch"
+
+    ###
+    ### keybase-bin
+    ###
+
+    keybase_bin_repo="$build_root/arch/keybase-bin"
+    clone_maybe "$git_url" "$keybase_bin_repo"
+
+    cp "$here/keybase.install" "$keybase_bin_repo"
+
+    # Arch doesn't allow dashes in its version numbers.
+    pkgver="$(cat "$build_root/VERSION" | sed s/-/_/)"
+
+    # Debian replaces the + with a . to avoid URL mangling.
+    # Note that we've duplicated this logic in the PKGBUILD, to make our diffs
+    # nicer, as per "Eschwartz commented on 2017-07-28 20:41" at
+    # https://aur.archlinux.org/packages/keybase-bin.
+    debver="$(cat "$build_root/VERSION" | sed s/+/./)"
+
+    deb_i386="$(ls "$build_root"/deb/i386/*.deb)"
+    sum_i386="$(sha256sum "$deb_i386" | awk '{print $1}')"
+
+    deb_amd64="$(ls "$build_root"/deb/amd64/*.deb)"
+    sum_amd64="$(sha256sum "$deb_amd64" | awk '{print $1}')"
+
+    if [ "$src_prefix" = "." ]; then
+        cp $deb_i386 $keybase_bin_repo/keybase_${debver}_i386.deb
+        cp $deb_amd64 $keybase_bin_repo/keybase_${debver}_amd64.deb
+    fi
```

Comment:
Maybe just move this block into the one caller that requires it?

Reply:
Yeah I didn't want to duplicate all the `debVer` logic above.  Is it safe to use `*`?  I know in my testing I tend to accumulte several debs in the build directory with different build times, seems like that would break.

Reply:
Oh woops, I meant to post these in a batch instead of spamming you. I think copying in a ton of debs will actually work fine, since the PKGBUILD script specifies the full name of the one it's opening. But for sure if it ends up being dirty to move it, we can keep it as it is.

Reply:
Oh but the tricky part is changing the format of the debian version string, right?  The version is `$deb_i386` isn't in `$debver` format, as far as I remember.

Reply:
Not for the version of the _destination_ file though, right?  And it would be an error to copy multiple files (if the `*` expanded) to a single destination file anyway.

Reply:
No, because we are changing the actual filename, to translate the version into a different format.  Just moving it to a different directory isn't good enough, as far as I can tell.

Reply:
Gotcha.